src/dcap_provider.cpp

```diff
@@ -714,3 +721,112 @@ extern "C" sgx_plat_error_t sgx_ql_set_logging_function(
     logger_callback = logger;
     return SGX_PLAT_ERROR_OK;
 }
+
+extern "C" sgx_plat_error_t sgx_get_qe_identity_info(
+    sgx_qe_identity_info_t** pp_qe_identity_info)
+{
+    std::string intel_cert_base_url = INTEL_QE_IDENTITY_URL;
+    sgx_qe_identity_info_t *p_qe_identity_info = NULL;
+    char* buffer = nullptr;
+
+    if (!pp_qe_identity_info)
+    {
+        log(SGX_QL_LOG_ERROR, "Invalid parameter pp_qe_identity_info"); 
+        return SGX_PLAT_ERROR_INVALID_PARAMETER;
+    }
+
+    try 
+    {
+        std::vector<uint8_t> identity_info;
+        std::string issuer_chain;
+        size_t total_buffer_size = 0;
+
+        const auto curl = curl_easy::create(intel_cert_base_url, CYBERTRUST_ROOT_CERT);

```

Comment:
Intel does not use the cybertrust root. They seem to be using COMODO.

If this is working, then we have a bug with our TLS cert processing. Does this work? If so, let's open a bug and figure out why :)

Reply:
“Retrieve QE Identity” does not require any client authentication.

Reply:
This is server auth, not client auth.

Reply:
Replaced with the cert you provided.

Reply:
Undid the added cert.

Reply:
Please see my comments on the following issue:
https://github.com/Microsoft/Azure-DCAP-Client/issues/8

Reply:
We still need the correct certificate, even if issue #8 remains open.